tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuConverterConfig.java

```diff
@@ -0,0 +1,379 @@
+        return minimalDraftStatus;
+    }
+
+    @Override public Path getSpecialsDir() {
+        return specialsDir;
+    }
+
+    @Override public boolean emitReport() {
+        return emitReport;
+    }
+
+    // Currently hard-coded "hacks" which could be encoded via the builder if wanted.
+
+    @Override public Map<String, String> getForcedAliases(IcuLocaleDir dir) {
+        switch (dir) {
+        case COLL:
+            return ImmutableMap.<String, String>builder()
+                // It is not at all clear why this is being done (we expect "sr_Latn_ME" normally).
+                // TODO: Find out and document this properly.
+                .put("sr_ME", "sr_Cyrl_ME")
+
+                // This appears to be a hack to avoid needing to copy and maintain the same "zh"
+                // data for "yue". The files for "yue" in this directory should be empty otherwise.
+                //
+                // The maximized versions of "yue_Hans" is "yue_Hans_CN" (vs "zh_Hans_CN"), and for
+                // "yue" it's "yue_Hant_HK" (vs "zh_Hant_HK"), so the aliases are effectively just
+                // rewriting the base language.
+                .put("yue_Hans", "zh_Hans")
+                .put("yue", "zh_Hant")
+                .build();
+        case RBNF:
+            // It is not at all clear why this is being done. It's certainly not exactly the same
+            // as above, since (a) the alias is reversed (b) "zh_Hant" does exist, with different
+            // data than "yue", so this alias is not just rewriting the base language.
+            // TODO: Find out and document this properly.
+            return ImmutableMap.of("zh_Hant_HK", "yue");
+        default:
+            return ImmutableMap.of();
+        }
+    }
+
+    // This set of locale files in each directory denotes the supported/available locales for that
+    // API. In most cases, it's the same set, but a few directories support only a subset of IDs.
+    @Override public ImmutableSet<String> getTargetLocaleIds(IcuLocaleDir dir) {
+        switch (dir) {
+        case COLL:
+            return COLL_LOCALE_IDS;
+        case BRKITR:
+            return BRKITR_LOCALE_IDS;
+        case RBNF:
+            return RBNF_LOCALE_IDS;
+        default:
+            return ICU_LOCALE_IDS;
+        }
+    }
+
+    // The primary set of locale IDs to be generated. Other, directory specific, sets should be
+    // subsets of this. Some of these ID are aliases, so XML files may not exist for all of them.
+    //
+    // This was further modified (in order to better match the set of generated ICU files) by:
+    // * Removing "es_003" (which just seems to be ignored in current code)
+    // * Adding:  "en_NH", "sr_XK", "yue_CN", "yue_HK" (deprecated locale IDs in the manual config)
+    // * Adding: "no_NO_NY" (a not even structurally valid ID that exists for very legacy reasons)
+    private static final ImmutableSet<String> ICU_LOCALE_IDS = ImmutableSet.of(
```

Comment:
no... please remove this list.  This should be generated from CLDR data + configuration. I strongly object to a static list in this file!

Reply:
Oh I'm very against this, but this list is not trivially computable from the CLDR data (or not by any way I know). When I had this just make all locales, you get about 20 or 30 more locales.

So, if you want this to go away, tell me how to get exactly this list from CLDR data (since, at least for migration of tools) there's a strong requirement to produce identical results.

Reply:
Do you mean `icu-config.xml`?  If that's what this is replacing, please put this in a properties file (or equivalent), and make it a command line parameter, not baked into the source.

Reply:
Yes, that's what's being replaced. I can make a config/properties file, but so far nobody has suggested it's important for it to be easily configurable. Is there a use case other than "generate all the ICU data files" that would need a different list?

Review the rest (or as much as you want) and I'll come back to this tomorrow (I have to be in the hospital for at least some of tomorrow, so I might not get to it until later).

Reply:
ok, take care. i will review the rest.

> Is there a use case other than "generate all the ICU data files" that would need a different list?

yes. "I only want these files" or "I want to include these files from seed".

Reply:
I've updated the build stuff and removed the API jar, but I haven't done the config stuff yet (and I'd rather do that once everything else is in since I'd like a longer discussion about use cases and config). For example if I could determine the list via examining the CLDR data, all the better.

Reply:
Given the discussion started in https://docs.google.com/document/d/1QmpLD82RKBS0CDbWkIQaWFd80Ne_XiEtXBK88n9a5Yg/edit#
which needs people to look at it and come to a concensus, I'm strongly in favour of not letting this specific detail block the initial submission. There's a bunch more thinking and work around the config stuff to be done, but it's not important if it's done now or in a much small targeted PR in a week or so.

Reply:
OK- i can see that.